packages/apps/graph-client/src/components/graphql-query-dialog/graphql-query-dialog.tsx

```diff
@@ -0,0 +1,44 @@
+import { Button, Dialog, DialogContent, Divider } from '@kadena/react-ui';
+import type { DocumentNode } from 'graphql';
+import { print } from 'graphql';
+import React, { useState } from 'react';
+
+interface IGraphQLQueryDialog {
+  queries: DocumentNode[];
+  variables: Record<string, unknown>;
+}
+
+export const GraphQLQueryDialog = (props: IGraphQLQueryDialog): JSX.Element => {
+  const { queries, variables } = props;
+
+  const [isOpen, setIsOpen] = useState(false);
+
+  return (
+    <>
+      <Button
+        icon="Information"
+        title="Show the GraphQL query used."
+        variant="compact"
+        onClick={() => setIsOpen(true)}
+      >
+        GraphQL
+      </Button>
+      <Dialog
+        isOpen={isOpen}
+        onOpenChange={(isOpen: boolean) => setIsOpen(isOpen)}
+      >
+        {() => (
+          <DialogContent>
+            <p>Query</p>
+            {queries.map((query, index) => (
```

Comment:
If there's more than one query with variables, does it still make sense to represent them in the same `<DialogContent>`. My concern is only if the variables happen to have the same name, it can be confusing to determine which one belongs to which.

Reply:
Yep you're right. I have redone it to make more sense. Let me know what you think!

![chrome_1vc8fPoF9Y](https://github.com/kadena-community/kadena.js/assets/6686385/10c20914-d08c-467d-86bb-1770edc7e829)